tests/test_input.py

```diff
@@ -10,2690 +10,2624 @@
-        "logs": output_dir,
-        "results": output_dir,
-        "sampler": "RFBSampler",
-    }
-
-    config_template_path = config_dir / "test_config_parsing.template.cfg"
-    config_path = fill_in_config_options(config_template_path, values_to_fill_dict, "bad_sampler")
-
-    yield check_config_parsing_value_error, config_path
-
-
-def test_config_parsing_bad_hashing():
-    """Test that config parsing raises error with `feature_hasher` but not `hasher_features`."""
-    # make a simple config file that has bad feature hashing information
-    values_to_fill_dict = {
-        "experiment_name": "config_parsing",
-        "task": "evaluate",
-        "train_directory": train_dir,
-        "test_directory": test_dir,
-        "featuresets": "[['f1', 'f2', 'f3']]",
-        "learners": "['LogisticRegression']",
-        "logs": output_dir,
-        "results": output_dir,
-        "feature_hasher": "True",
-    }
-
-    config_template_path = config_dir / "test_config_parsing.template.cfg"
-    config_path = fill_in_config_options(config_template_path, values_to_fill_dict, "bad_hashing")
-
-    yield check_config_parsing_value_error, config_path
-
-
-def test_config_parsing_bad_featuresets():
-    """Test that config file parsing raises error with badly specified featuresets."""
-    # make a simple config file that has bad feature sets
-    values_to_fill_dict = {
-        "experiment_name": "config_parsing",
-        "task": "evaluate",
-        "train_directory": train_dir,
-        "test_directory": test_dir,
-        "learners": "['LogisticRegression']",
-        "logs": output_dir,
-        "results": output_dir,
-    }
-
-    for featuresets, sub_prefix in zip(
-        [None, "[]", "{'f1', 'f2', 'f3'}", "[['f1', 'f2'], 'f3', 'f4']"],
-        ["no_feats", "empty_feats", "non_list_feats1", "non_list_feats2"],
-    ):
-        if featuresets is not None:
-            values_to_fill_dict["featuresets"] = featuresets
+
+        config_path = fill_in_config_options(
+            config_template_path, values_to_fill_dict, "no_objectives_learning_curve"
+        )
+
+        (
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            do_grid_search,
+            grid_objectives,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            _,
+            output_metrics,
+            _,
+        ) = parse_config_file(config_path)
```

Comment:
ok... i know this is from the pre-commit, but yuck...  lol

Reply:
i feel like it might actually be more readable for these to do something like:
```
ph = parse_config_file(config_path)
do_grid_search = ph[14]
grid_objectives = ph[15]
output_metrics = ph[-2]
```

Reply:
Good call! Definitely less yucky :)